tests/model_registry/rbac/conftest.py

```diff
@@ -89,6 +92,77 @@ def sa_token(service_account: ServiceAccount) -> str:
         raise
 
 
+@pytest.fixture(scope="function")
+def new_group(
+    request: pytest.FixtureRequest,
+    admin_client: DynamicClient,
+    test_idp_user_session: UserTestSession,
+) -> Generator[str, None, None]:
+    """
+    Fixture to create a new OpenShift group and add a user, then delete the group after the test.
+    The group name is passed as a parameter to the fixture and the user is taken from test_idp_user_session.
+    """
+
+    group_name = request.param
+    with Group(
+        client=admin_client,
+        name=group_name,
+        users=[test_idp_user_session.username],
+        wait_for_resource=True,
+    ) as _:
+        LOGGER.info(f"Group {group_name} created successfully.")
+        yield group_name
+        LOGGER.info(f"Group {group_name} deletion initiated by context manager.")
```

Comment:
this would be a good candidate for factory/composition, I assume we will need a fixture to create an empty group at some point, and that could be the factory/builder, while your fixture could pass in the additional user. Maybe worth keeping in mind for some refactoring.